specifyweb/frontend/js_src/lib/components/QueryBuilder/Export.tsx

```diff
@@ -86,7 +86,15 @@ export function QueryExportButtons({
     );
 
     const joinedSelected = selectedResults?.map((subArray) =>
-      subArray?.slice(1).join(separator)
+      subArray
+        ?.map((item) => {
+          if (typeof item === 'string') {
+            return item?.replace(new RegExp(separator, 'g'), '-');
```

Comment:
hmm, I think there is no need for regex.
simply use `.replaceAll(separator, '-') instead

Reply:
if you already checked for `typeof item === 'string'`, why have a question mark here?

really `?.` should be sufficient - you shouldn't need the `if()` or the other return, or the `{` `}`

Reply:
I added the 'if' for: "Property 'replaceAll' does not exist on type 'number'."

Reply:
ah, I see now what this code is trying to do.
replacing occurrences of separators with a different character is not correct - that corrupts the data. Fortunately, a better solution exists. See:
https://stackoverflow.com/questions/4617935/is-there-a-way-to-include-commas-in-csv-columns-without-breaking-the-formatting

Reply:
even better, you don't need to do this on your own, because it's bug prone (there are other edge cases to consider - unicode, encodings, new lines...)
that is why we use a library `csv-stringify` for that
the `downloadDataSet` function could probably be modified/refactored to suit your needs